Xamarin.Essentials/TextToSpeech/TextToSpeech.android.cs

```diff
@@ -0,0 +1,224 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Android.OS;
+using Android.Speech.Tts;
+using AndroidTextToSpeech = Android.Speech.Tts.TextToSpeech;
+
+namespace Xamarin.Essentials
+{
+    public static partial class TextToSpeech
+    {
+        const int maxSpeechInputLengthDefault = 4000;
+
+        static WeakReference<TextToSpeechImplementation> textToSpeechRef = null;
+
+        static TextToSpeechImplementation GetTextToSpeech()
+        {
+            if (textToSpeechRef == null || !textToSpeechRef.TryGetTarget(out var tts))
+            {
+                tts = new TextToSpeechImplementation();
+                textToSpeechRef = new WeakReference<TextToSpeechImplementation>(tts);
+            }
+
+            return tts;
+        }
+
+        internal static async Task PlatformSpeakAsync(string text, SpeakSettings settings, CancellationToken cancelToken = default)
+        {
+            using (var textToSpeech = GetTextToSpeech())
```

Comment:
Do we really need to dispose before each of these? Any advantage? Will need to re-init each time...